File: Front/src/model/ic/ic.ts
import {icPropriedades} from 'src/model/ic/propriedades';
import {corpTipo} from 'src/model/corp/tipo';
import {segOrganograma} from 'src/model/seg/organograma';

export interface icIc{
    id: number;
    idPai: number;
    nome: string;
    ativo: boolean;
    ativoFinal: boolean;
    ativoPai: boolean;
    propriedades: icPropriedades[];
    idTipo: number;
    idOrganograma: number;
    nomeCompleto: string;
    listaAncestrais: string;
    nivel: number;
    observacao: string;
    lstAncestrais: number[];
 
    pai: icIc;
    filhos: icIc[];
    ancestrais: icIc[];
    tipo: corpTipo;
    responsavel: segOrganograma;
    // Conhecimentos: ICConhecimento[];
    // Segredos: ICSegredo[];
    // Dependentes: ICDependencia[];
    // Dependencias: ICDependencia[];

}

File: MCP/DTOs/IC/IC.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Reflection;
using System.Text;

namespace McpServer.DTOs.IC;

public record IC
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("idpai")]
    public int? IdPai { get; set; }

    [Column("nome")]
    public string Nome { get; set; } = string.Empty;

    [Column("ativo")]
    public bool Ativo { get; set; }

    [Column("propriedades", TypeName = "jsonb")]
    public List<ICPropriedade>? Propriedades { get; set; }

    [Column("idtipo")]
    public int IdTipo { get; set; }

    [Column("idorganograma")]
    public int? IdOrganograma { get; set; }



    [ForeignKey("IdPai")]
    public IC? Pai { get; set; }

    [ForeignKey("IdTipo")]
    public DTOs.Corp.Tipo? Tipo { get; set; }

    [ForeignKey("IdOrganograma")]
    public Seg.Organograma? Responsavel { get; set; }


    public void Altera(IC item)
    {
        this.Nome = item.Nome;
        this.Ativo = item.Ativo;
        this.IdTipo = item.IdTipo;
        this.Propriedades = item.Propriedades;

    }


}
